src/compiler/types.ts

```diff
@@ -1743,10 +1743,12 @@ namespace ts {
         FromSignature           = 0x00040000,  // Created for signature assignment check
         ObjectLiteral           = 0x00080000,  // Originates in an object literal
         /* @internal */
-        ContainsUndefinedOrNull = 0x00100000,  // Type is or contains Undefined or Null type
+        FreshObjectLiteral      = 0x00100000,  // Fresh object literal type

```

Comment:
Yup, I think we should give up the first assignment bit. I think it is a strange rule.